socketIO_client/transports.py

```diff
@@ -150,16 +155,19 @@ def recv_packet(self):
             raise ConnectionError('recv disconnected (%s)' % e)
         except SocketError as e:
             raise ConnectionError('recv disconnected (%s)' % e)
-        if not isinstance(packet_text, six.binary_type):
-            packet_text = six.b(packet_text)
-        engineIO_packet_type, engineIO_packet_data = parse_packet_text(
-            packet_text)
-        yield engineIO_packet_type, engineIO_packet_data
+        if packet_text:
+            if not isinstance(packet_text, six.binary_type):
+                packet_text = six.b(packet_text)
+            engineIO_packet_type, engineIO_packet_data = parse_packet_text(
+                packet_text)
+            yield engineIO_packet_type, engineIO_packet_data
 
     def send_packet(self, engineIO_packet_type, engineIO_packet_data=''):
         packet = format_packet_text(engineIO_packet_type, engineIO_packet_data)
         try:
+            self.lock.acquire()
             self._connection.send(packet)
+            self.lock.release()
```

Comment:
Depends, does _connection.send raise exceptions that are somehow recoverable, probably make sense to do in finally though